tools/packaging/kernel/configs/fragments/common/fs.conf

```diff
@@ -52,8 +52,6 @@ CONFIG_AIO=y
 # Docker in Docker support requires overlay
 CONFIG_OVERLAY_FS=y
 CONFIG_OVERLAY_FS_INDEX=y
-CONFIG_OVERLAY_FS_REDIRECT_DIR=y
```

Comment:
@skaegi This was added to support docker-in-docker use case. According to the docs, you can still disable this with `"redirect_dir=off" ` module option. Maybe you can consider passing this on the kernel command line and see if that works out.

Reply:
Looking back at #1075 I only suggested `CONFIG_OVERLAY_FS_METACOPY` be added. I'm not sure why `CONFIG_OVERLAY_FS_REDIRECT_DIR` would ever be required for dind. We certainly don't use it in our kernel and do literally millions of dind builds a month.

Reply:
More info here -- https://dev.to/jarekprzygodzki/a-curious-case-of-slow-docker-image-builds-2o7k

Reply:
Thanks for the links @skaegi. 
This is not related to Kata itself or `dind` usecase, but I do recall running into an issue where files in renamed directories were not correctly propogated to the upper layer in overlay file system due to `CONFIG_OVERLAY_FS_REDIRECT_DIR` not being enabled in the host system on a particular linux distro. Its been a while though, and I am not aware of changes since then in this space to not require this option anymore nor dabbled in dinds recently.
If you feel it is safe to turn off this option, lets do that. 

I just happened to take a look at the kernel change related to this kernel option: https://github.com/torvalds/linux/commit/d47748e5ae5af6572e520cc9767bbe70c22ea498#diff-315c61b86e39b9b47f4ab0cd9efc2467
That commit suggests that for metacopy to be turned on for overlay file system, it depends on `redirect_dir` to be turned on. Referring to this line in the commit message:  `This patch makes metacopy=on imply redirect_dir=on.`